task/export.go

```diff
@@ -61,24 +63,75 @@ func TaskExport(tctx *TaskContext) (*TaskHandlerOutput, error) {
 	defer cancel()
 	tctx = tctx.WithContext(ctx)
 	content := tctx.Progress.TrackReader(file.Body, size, 1)
-	output, err := uploadFile(tctx, asset, content)
+
+	name := "asset-" + asset.PlaybackID
+	contentType := "video/" + asset.VideoSpec.Format
+	cid, internalMetadata, err := uploadFile(tctx, customParams, ipfsParams, name, content, contentType)
+	if err != nil {
+		return nil, err
+	}
+	nftCid, err := uploadNftMetadata(tctx, ipfsParams, cid)
+	if err != nil {
+		return nil, err
+	}
+	return &TaskHandlerOutput{
+		TaskOutput: &data.TaskOutput{
+			Export: &data.ExportTaskOutput{
+				Internal: internalMetadata,
+				IPFS: &data.IPFSExportInfo{
+					VideoFileCID:   cid,
+					NFTMetadataCID: nftCid,
+				},
+			},
+		},
+	}, nil
+}
+
+func TaskExportData(tctx *TaskContext) (*TaskHandlerOutput, error) {
+	var (
+		params       = tctx.Params.ExportData
+		customParams = params.Custom
+		ipfsParams   = params.IPFS
+	)
+	content := strings.NewReader(params.Content)
+	name := fmt.Sprintf("%s-%s", params.Type, params.ID)
+	contentType := "application/json"
```

Comment:
If we only support JSON, might want to make the `params.Content` a `json.RawMessage` instead. We'd avoid encoding the JSON as a string as well as validate that the data is actually a JSON (while the type itself is just an alias for `[]byte`, can use `bytes.NewReader` as well)

Reply:
Updated in a6b5ea2c88768cca6a3e3aaca2f861ea68fec877